src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core/Credential/ICredentialService.cs

```diff
@@ -0,0 +1,14 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using System.Threading;
+using System;
+
+namespace Microsoft.Performance.Toolkit.PluginManager.Core.Credential
+{
+    public interface ICredentialService
+    {
+        Task<ICredentials> GetCredentialsAsync(Uri uri, CancellationToken cancellationToken);
```

Comment:
Why is this restricted to always being a Uri?

Reply:
Providers are updated to take in generic plugin sources